### ISSUE TITLE
Android app crashes after update for the first time

### ISSUE THREAD

**Issue by szotsaki** [nextcloud/android#11275]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1.  Start the Android application

### Expected behaviour

 App starts

### Actual behaviour

Application crashes right on start after upgrading. Second start launches it normally.

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'int java.util.ArrayList.size()' on a null object reference
    at com.owncloud.android.ui.fragment.ExtendedListFragment.restoreIndexAndTopPosition(ExtendedListFragment.java:453)
    at com.owncloud.android.ui.fragment.OCFileListFragment.onBrowseUp(OCFileListFragment.java:891)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onBackPressed(FileDisplayActivity.java:1088)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onOptionsItemSelected(FileDisplayActivity.java:789)
    at android.app.Activity.onMenuItemSelected(Activity.java:4442)
    at androidx.activity.ComponentActivity.onMenuItemSelected(ComponentActivity.java:514)
    at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:265)
    at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:266)
    at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:110)
    at androidx.appcompat.widget.ToolbarWidgetWrapper$1.onClick(ToolbarWidgetWrapper.java:188)
    at android.view.View.performClick(View.java:7467)
    at android.view.View.performClickInternal(View.java:7444)
    at android.view.View.access$3700(View.java:840)
    at android.view.View$PerformClick.run(View.java:28830)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:329)
    at android.app.ActivityThread.main(ActivityThread.java:8063)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30230190`
* Build flavor: `generic`

### Device information
* Brand: `motorola`
* Device: `nio`
* Model: `moto g(100)`
* Id: `S1RTS32.41-20-16-1-5`
* Product: `nio_retaile`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `d6673-9e875`

### Android version

12.0

### Device brand and model

-

### Stock or custom OS?

Stock

### Nextcloud android app version

3.23.1

### Nextcloud server version

25.0.2

### Using a reverse proxy?

No

**Comment by AlertSubject** [comment]
Having the same issue

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
    at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:910)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByPath(FileDataStorageManager.java:138)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByEncryptedRemotePath(FileDataStorageManager.java:122)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByPath(FileDataStorageManager.java:118)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onStart(FileDisplayActivity.java:2423)
    at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1529)
    at android.app.Activity.performStart(Activity.java:8420)
    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3889)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2466)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:240)
    at android.os.Looper.loop(Looper.java:351)
    at android.app.ActivityThread.main(ActivityThread.java:8321)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1012)

```

### App information
* ID: `com.nextcloud.android.beta`
* Version: `20230125`
* Build flavor: `versionDev`

### Device information
* Brand: `OnePlus`
* Device: `OnePlus9Pro`
* Model: `LE2123`
* Id: `TP1A.220624.014`
* Product: `OnePlus9Pro_EEA`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `R.d0e58f_17c-2`

**Comment by rtbentley** [comment]
I'm not entirely certain this is exactly the same error, but it is a crash of the Android app immediately when it starts.

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
    at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:907)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByPath(FileDataStorageManager.java:138)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByEncryptedRemotePath(FileDataStorageManager.java:122)
    at com.owncloud.android.datamodel.FileDataStorageManager.getFileByPath(FileDataStorageManager.java:118)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onStart(FileDisplayActivity.java:2423)
    at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1435)
    at android.app.Activity.performStart(Activity.java:8134)
    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3554)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7886)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

```

### App information
* ID: `com.nextcloud.android.beta`
* Version: `20230121`
* Build flavor: `versionDev`

### Device information
* Brand: `motorola`
* Device: `kievv`
* Model: `motorola one 5G UW ace`
* Id: `RRVS31.Q3-15-3-2-2`
* Product: `kievv_vzw`

### Firmware
* SDK: `30`
* Release: `11`
* Incremental: `788b31`

**Comment by AlertSubject** [comment]
Latest nightly build fixes the crash for me
https://download.nextcloud.com/android/dev/

**Comment by regs01** [comment]
There are seem to be no updates in F-Droid since January 21.

**Comment by regs01** [comment]
I noticed 20230126 wasn't installed. It fails to install and stays on 20230121. And I can't delete and clean install it, because resyncing large folder is still very problematic.

**Comment by AlvaroBrey** [comment]
@AlertSubject @rtbentley your issue is different: #11305 and should be fixed by latest dev versions

**Comment by rtbentley** [comment]
@AlvaroBrey Thank you.  It is indeed resolved.  Thank you for pointing me to the correct ticket.

**Comment by joshtrichards** [comment]
Fixed in #13084
